lib/kmts.rb

```diff
@@ -191,13 +191,23 @@ def generate_query(type, data, id = nil)
       data.update('_k' => @key)
       data.update '_d' => 1 if data['_t']
       data['_t'] ||= Time.now.to_i
-      
+
       unsafe = Regexp.new("[^#{URI::REGEXP::PATTERN::UNRESERVED}]", false, 'N')
       
       data.inject(query) do |query,key_val|
         query_arr <<  key_val.collect { |i| URI.escape(i.to_s, unsafe) }.join('=')
       end
       query = '/' + type + '?' + query_arr.join('&')
+
+      begin
+        unless data['_t'].to_s =~ /\d{10}/ and data['_t'].to_i >= 0
```

Comment:
Could you factor this validation logic into a separate method?  The exception only to be caught locally seems a bit ungainly, though I understand it's necessary given `#log_error`.

Otherwise, note your regular expression pattern is unanchored and arguably a bit overspecified; I'd suggest `/\A\d+\z/` instead perhaps.  It's also necessary to return after handling the exception, as otherwise the tracking request is still issued.  Finally as a very minor style nit, our guide advises against `and` in favor of `&&`.